### PR TITLE
Emit a warning for `around(:context)`

### LINKED ISSUE #2486: `around(:context)` does not work as expected but does not provide a warning
We only support `around(:example)` hooks but `around(:context)` hooks do not raise an exception or print a warning or anything -- instead, they act just like an `around(:example)` hook.  Example:

``` ruby
RSpec.describe "Around" do
  around(:context) do |ex|
    puts "around before"
    ex.run
    puts "around after"
  end

  it 'works' do
    puts "in example 1"
  end

  it 'works again' do
    puts "in example 2"
  end
end
```

This produces:

```
Around
around before
in example 2
around after
  works again
around before
in example 1
around after
  works
```

See #1598 for some background on this.

### PR DESCRIPTION
**Pull request by pirj**
fixes #2486

related to #1031

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Bug Fixes:
 * Prevent bisect command from blocking when number of specs exceeds file
   descriptor limit on OSX or Linux. (Benoit Tigeot, #2669)
 * Prevent warnings being issued on Ruby 2.7.0. (Jon Rowe, #2680)
+* Emit a warning when `around` hook is used with `:context` scope
+  (Phil Pirozhkov, #2687)
```
Comment (JonRowe) [inline Changelog.md]: Changelog in the wrong place.

**Comment by JonRowe** [comment]
👍

**Comment by shepmaster** [comment]
One thing confusing about this warning is that it points to something that *isn't* literally `around(:context)`:

```
WARNING: `around(:context)` hooks are not supported and behave like `around(:example). Called from ...spec/controllers/clients_controller_spec.rb:31:in `block (2 levels) in <top (required)>'.
```

```
% sed -n 31p spec/controllers/clients_controller_spec.rb
    around(:all) do |example|
```

**Comment by JonRowe** [comment]
`:all` is an alias for `:context`

**Comment by shepmaster** [comment]
Sure, I'm not saying that it's incorrect, just *confusing*.

**Comment by JonRowe** [comment]
Just pointing out it is literally *is* `:context` to rspec.

**Comment by pirj** [comment]
We can probably move this warning deeper, to `scope_and_options_from` where we still have the original scope argument before normalization.

@shepmaster Would you want to tackle that?

**Comment by shepmaster** [comment]
> Would you want to tackle that?

Thank you for the offer, but it seems unlikely I'll have the time for it 😿.

We only had one case of this warning so *my* specific pain has passed. I mostly thought to post this in case other people have the same pain, or if y'all start getting a lot of reports.

**Pull request merged**